api/workloads/v1alpha1/helper.go

```diff
@@ -37,14 +37,32 @@ func (rbg *RoleBasedGroup) GetGroupSize() int {
 }
 
 func (rbg *RoleBasedGroup) GetWorkloadName(role *RoleSpec) string {
-	return fmt.Sprintf("%s-%s", rbg.Name, role.Name)
+	if rbg == nil {
+		return ""
+	}
+
+	workloadName := fmt.Sprintf("%s-%s", rbg.Name, role.Name)
+
+	// Kubernetes name length is limited to 63 characters
+	if len(workloadName) > 63 {
+		workloadName = workloadName[:63]
+		workloadName = strings.TrimRight(workloadName, "-")
+	}
+	return workloadName
 }
 
+// GetServiceName Because ServiceName needs to follow DNS naming conventions,
+// which do not allow names to start with a number. Therefore, the s- prefix
+// is added to the service name to meet this requirement.
 func (rbg *RoleBasedGroup) GetServiceName(role *RoleSpec) string {
-	if len(rbg.Name) > 0 && unicode.IsDigit(rune(rbg.Name[0])) {
-		return fmt.Sprintf("s-%s-%s", rbg.Name, role.Name)
+	svcName := fmt.Sprintf("s-%s-%s", rbg.Name, role.Name)
```

Comment:
The change introduces a new service name format (s-<rbg.Name>-<role.Name>) instead of reusing the workload name (<rbg.Name>-<role.Name>).  This breaks backward compatibility because Existing Services created by prior versions will no longer be recognized by the new code. The reconciler will attempt to query Services with the svc- prefix, fail with IsNotFound, and recreate them, potentially deleting old Services unintentionally.

How about checking if the rbg object is created by old version controller?

```golang
func (rbg *RoleBasedGroup) GetServiceName(role *RoleSpec) string {
    oldName := fmt.Sprintf("%s-%s", rbg.Name, role.Name)
    if exists, _ := CheckServiceExists(oldName); exists {
        return oldName
    }
    return fmt.Sprintf("s-%s-%s", rbg.Name, role.Name)
}
```